### ISSUE TITLE
Ignores **/venv/** successfully but not **/Lib/** (Neovim)

### ISSUE THREAD

**Issue by woertsposzibllen4me** [DetachHead/basedpyright#725]
Cannot figure out how to tell basedpyright to ignore Lib modules

pyproject.toml:
```
[tool.basedpyright]
ignore = [
  "**/venv/**",            # works
  "**/Lib/**",             # doesn't
]
```

Tried brute forcing many possible variations with no sucess. I know the `.toml` is picked up on (works for other options)

**Comment by DetachHead** [comment]
it should be ignoring those automatically. where is your venv located relative to your project?

**Comment by avario-cpu** [comment]
> it should be ignoring those automatically. where is your venv located relative to your project?

well venv is on root, and is ignored but not automatically.
lib is on Appdata\Local etc... will try specifiyng the root path because I haven't thought of using the drive letter altough it's a bit weird
but even `ignore = [""C:\\Users\\YourUsername\\AppData\\Local\\Programs\\Python\\Python312\\Lib"]` will not work

(forgot to specify I was on windows)

**Comment by DetachHead** [comment]
what if you use `exclude` instead of `ignore`?

**Comment by avario-cpu** [comment]
> what if you use `exclude` instead of `ignore`?
No luck either, altough I never really know what semantics I've tried, but I've been inserting variation of those path for the past 2 hours 😅.

**Comment by DetachHead** [comment]
hmm, some more questions:

- what commands are you using to create your venv?
- are you using the basedpyright CLI (if so, what command are you running exactly and is it in the same working directory as your `pyproject.toml`)?
- are you using the language server? if so which editor are you using?

**Comment by avario-cpu** [comment]
> what commands are you using to create your venv?

create ? `venv venv` and then `venv\scripts\activate`

> are you using the basedpyright CLI (if so, what command are you running exactly and is it in the same working directory as your pyproject.toml)?
> are you using the language server? if so which editor are you using?
> ...

Lang serv. Nvim. All defaults.  installed version 1.18.2  (Using Mason)

**Comment by DetachHead** [comment]
ah, usually i call my venv `.venv` which (based)pyright automatically excludes. thanks for the info, will investigate this

**Comment by avario-cpu** [comment]
My venv ignore is not an issue at all tho ^^ its the Lib modules I can't figure out how to ignore

**Comment by DetachHead** [comment]
oh right. in that case i have no clue what it could be lol. are you able to try the CLI and if possible, [another editor](https://docs.basedpyright.com/latest/installation/ides/) just to see if the issue is specific to neovim?

**Comment by avario-cpu** [comment]
Yeah I'll have a try with CLI and see but I need to do something else for a while and comeback cause I'm burned out rn

**Comment by avario-cpu** [comment]
I can say that running basedpyright "C:\Users\ville\AppData\Local\Programs\Python\Python312\Lib\dataclasses.py"

with

```
ignore = [
  "**/venv/**",
  "../../../AppData/Local/Programs/Python/Python312/Lib/**",
] 
```

in the pyproject does provide diagnostics, wich i think it shouldn't ?

I'm pretty confused by the path difference and don't really undsertand how it works tbh, did not seem to be able to pass an `--ingore` or `ignorepattern` in the cli command itself, maybe doing it wrong idk, I'd need steps by steps to really know what I'm doing

**Comment by DetachHead** [comment]
oh yeah i see now. not sure why that's not working, but either way it should ignore any errors from the standard library by default

**Comment by DetachHead** [comment]
this was partially addressed upstream. in the next release you can specify an absolute path. but i'll leave this issue open as it's not a full solution to the problem. it should ignore standard library modules by default

**Comment by avario-cpu** [comment]
I report with great shame that a simple change to `ignore = ["../**/Lib/**", "**/venv/**"]` managed to fix the issue. I thought  that I would need a set of `../` per levels up the directory until the root disk, hence why the past attempts did not work...

On the rare chance of some other noob having the same issue as me:

in the case of your pyproject being in `C:/Users/Username/Project`

```
ignore = [
  "**/venv/**",
  "../../../AppData/Local/Programs/Python/Python312/Lib/**", # BAD
] 
```

```
ignore = [
  "**/venv/**",
  "../AppData/Local/Programs/Python/Python312/Lib/**", # GOOD
] 
```

**Comment by DetachHead** [comment]
thanks for the update. i'm gonna leave this issue open though because you shouldn't have to manuallyu configure this in the first place. it should ignore the standard library by default

**Comment by avario-cpu** [comment]
Can't figure out how to ignore Lib directories again after my previous solution stopped working.. aergh

**Comment by bro69noscope** [comment]
So I would like to come back to this to close and maybe help another beginner ...

After having some struggles again but this time setting up `ruff`... I realized my problem is likely just about a NeoVim "issue" (it's probably working as intended really) and I dont think basedpyright has much to do with it.

Buffer in Neovim don't operate the same way most code editors "tabs" do, when going to definition on an imported symbol, the nvim goes to the absolute path of that module and open the file in a buffer with the current neovim configuration, if lsp server are setup, they might run into that new "workspace" unless explicitely told not to do so... which I have no idea how to do, because nvim can be complicated.

In short, rules from a project's root `pyproject.toml` are likely to apply only to that directory for NeoVim LSPs (assuming you use a popular setup with [lspconfig.nvim](https://github.com/neovim/nvim-lspconfig) ) therefore, you can only exclude files which are childs of that `pyproject` root folder.

I have no idea how I still originally managed to ignore my Lib directory from the `toml`, but we can just cast that as unreliable and not the way to go.

# **TLDR for the whole Issue:** 
Any exclusion outside the project root (typically StdLib, but not .venv) from the typical NeoVim setup has to be done on the lspconfig side of things and and not `pyproject.toml` :

```lua
local lspconfig = require("lspconfig")

      lspconfig.ruff.setup({
        init_options = {
          settings = {
            exclude = {
              "/usr/lib/python*/**",
            },
          },
        },
      })

      lspconfig.basedpyright.setup({
        settings = {
          basedpyright = {
            analysis = {
              ignore = {
                "/usr/lib/python*/**",
              },
            },
          },
        },
      })

```

**Comment by DetachHead** [comment]
thanks for the update, i still think it's an issue that errors from the standard library are not ignored by default (which happens in vscode as well) but since this seems like a neovim-specific issue, i'll open a separate issue for that
